test/functions/generic/generic-formal-qs.chpl

```diff
@@ -0,0 +1,13 @@
+record R { type t; }
+proc a(arg: R) { } // warning asking this to be written as 'arg: R(?)'
+proc b(arg: R(?)) { }
+a(new R(int));
+b(new R(int));
+
+proc getGenericType() type {
+  return R(?);
+}
+proc c(arg: getGenericType()) { }
+proc d(arg: getGenericType()(?)) { }
```

Comment:
I also think this looks ugly, but again, not enough to backtrack or hold things up :(